lib/wanda_web/views/execution_view.ex

```diff
@@ -58,4 +61,19 @@ defmodule WandaWeb.ExecutionView do
 
   defp map_checks_results(:running, _), do: nil
   defp map_checks_results(:completed, %{"check_results" => check_results}), do: check_results
+
+  defp count_critical(:running, _), do: nil
+
+  defp count_critical(:completed, %{"check_results" => check_results}),
+    do: Enum.count(check_results, &(&1["result"] == "critical"))
```

Comment:
Could we generalize this to

```elixir
defp count(%{"checks_results" => check_results}, result),
  do: Enum.count(check_results, &(&1["result"] == result))
```

?

Reply:
It looks a good idea to me. It is an internal function, so we shouldn't expect incorrect usage, and in the worst case, we would get a 0 count value

Reply:
yes I agree, I totally missed this!